rules/gcp_audit_rules/gcp_dns_zone_modified_or_deleted.py

```diff
@@ -0,0 +1,29 @@
+from panther_base_helpers import deep_get
+
+
+def rule(event):
+    methods = (
+        "dns.changes.create",
+        "dns.managedZones.delete",
+        "dns.managedZones.patch",
+        "dns.managedZones.update",
+    )
+    return deep_get(event, "protoPayload", "methodName", default="") in methods
+
+
+def title(event):
+    actor = deep_get(event, "protoPayload", "authenticationInfo", "principalEmail", default="")
+    method = deep_get(event, "protoPayload", "methodName", default="")
+    resource = deep_get(event, "protoPayload", "resourceName", default="")
+    return f"{actor} performed {method} on {resource}"
+
+
+def alert_context(event):
```

Comment:
I wonder if this would benefit from [gcp_base_helpers.gcp_alert_context()](https://github.com/panther-labs/panther-analysis/blob/master/global_helpers/gcp_base_helpers.py#L39)

Reply:
Oh nice 👀 That looks very comprehensive.